snakemake/executors.py

```diff
@@ -2052,3 +2059,605 @@ def run_wrapper(
             write_benchmark_records(bench_records, benchmark)
         except (Exception, BaseException) as ex:
             raise WorkflowError(ex)
+
+
+GoogleLifeScienceJob = namedtuple(
+    "GoogleLifeScienceJob", "job jobname jobid callback error_callback"
+)
+
+
+class GoogleLifeScienceExecutor(ClusterExecutor):
+    """the GoogleLifeSciences executor uses Google Cloud Storage, and
+       Compute Engine paired with the Google Life Sciences API.
+       https://cloud.google.com/life-sciences/docs/quickstart
+    """
+
+    def __init__(
+        self,
+        workflow,
+        dag,
+        cores,
+        envvars,
+        jobname="snakejob.{name}.{jobid}.sh",
+        printreason=False,
+        quiet=False,
+        printshellcmds=False,
+        container_image=None,
+        regions=None,
+        cache=False,
+        machine_type_prefix=None,
+        latency_wait=3,
+        local_input=None,
+        restart_times=None,
+        exec_job=None,
+        max_status_checks_per_second=1,
+    ):
+
+        # Attach variables for easy access
+        self.workflow = workflow
+        self.quiet = quiet
+        self.workdir = os.path.dirname(self.workflow.persistence.path)
+        self._save_storage_cache = cache
+        self._machine_type_prefix = machine_type_prefix
+
+        # Relative path for running on instance
+        snakefile = os.path.join(
+            self.workdir, os.path.basename(self.workflow.snakefile)
+        )
+
+        # Needs to be relative for commands run in worker (linux base)
+        self.snakefile = snakefile.replace(self.workdir + "/", "")
+        exec_job = (
+            "snakemake {target} --snakefile %s "
+            "--force -j{cores} --keep-target-files  --keep-remote "
+            "--latency-wait 0 "
+            "--attempt 1 {use_threads} "
+            "{overwrite_config} {rules} --nocolor "
+            "--notemp --no-hooks --nolock " % self.snakefile
+        )
+
+        # IMPORTANT: using Compute Engine API and not k8s == no support secrets
+        self.envvars = envvars or []
+
+        # Quit early if we can't authenticate
+        self._get_services()
+        self._get_bucket()
+
+        # Akin to Kubernetes, create a run namespace, default container image
+        self.run_namespace = str(uuid.uuid4())
+        self.container_image = container_image or "vanessa/snakemake:dev"
```

Comment:
As previously discussed let's try everything in order to not need a special container image.